Source/WebCore/css/parser/CSSPropertyParser.cpp

```diff
@@ -2534,10 +2536,39 @@ bool CSSPropertyParser::consumeListStyleShorthand(bool important)
     return m_range.atEnd();
 }
 
+bool CSSPropertyParser::consumeTextWrapShorthand(bool important)
+{
+    RefPtr<CSSValue> mode;
+    RefPtr<CSSValue> style;
+
+    for (unsigned propertiesParsed = 0; propertiesParsed < 2 && !m_range.atEnd(); ++propertiesParsed) {
+        if (!mode && (mode = CSSPropertyParsing::consumeTextWrapMode(m_range)))
+            continue;
+        if (!style && (style = CSSPropertyParsing::consumeTextWrapStyle(m_range)))
+            continue;
+        // If we didn't find at least one match, this is an invalid shorthand and we have to ignore it.
+        return false;
+    }
+
+    if (!m_range.atEnd())
+        return false;
+
+    // Fill in default values if one was missing from the multi-value syntax.
```

Comment:
What would happen if instead we only added the specified properties? Would a test fail? I guess I already forgot this from when I was doing CSS property work.

Reply:
We'd have no value to pass to `addProperty(` so we'd set the shorthand wouldn't do its job of resetting other longhand declarations. Tests cover this.